### ISSUE TITLE
SNOW-169193: Snowflake Exception has occurred: OperationalError 250003: Failed to execute request: 'SSLSocket' object has no attribute 'connection'

### ISSUE THREAD

**Issue by seoyeonberry** [snowflakedb/snowflake-connector-python#325]
Please answer these questions before submitting your issue. Thanks!

1. What version of Python are you using (`python --version`)?

**Python 3.7.2**

2. What operating system and processor architecture are you using (`python -c 'import platform; print(platform.platform())'`)?

- platform.architecture() -> '**64bit', 'WindowsPE**'
- platform.platform() -> **'Windows-10-10.0.18362-SP0**'

3. What are the component versions in the environment (`pip list`)? 

**alabaster==0.7.12
asn1crypto==0.24.0
astroid==2.2.5
astropy==3.2.1
atomicwrites==1.3.0
attrs==19.1.0
azure-common==1.1.23
azure-storage-blob==2.1.0
azure-storage-common==2.1.0
Babel==2.7.0
backcall==0.1.0
backports.functools-lru-cache==1.5
backports.os==0.1.1
backports.shutil-get-terminal-size==1.0.0
backports.tempfile==1.0
backports.weakref==1.0.post1
beautifulsoup4==4.7.1
bitarray==0.9.3
bkcharts==0.2
bleach==3.1.0
bokeh==1.2.0
boto==2.49.0
boto3==1.9.206
botocore==1.12.206
Bottleneck==1.2.1
certifi==2019.6.16
cffi==1.12.3
chardet==3.0.4
Click==7.0
cloudpickle==1.2.1
clyent==1.2.2
colorama==0.4.1
comtypes==1.1.7
conda==4.7.10
conda-build==3.18.8
conda-package-handling==1.3.11
conda-verify==3.4.2
contextlib2==0.5.5
cryptography==2.7
cycler==0.10.0
Cython==0.29.12
cytoolz==0.10.0
dask==2.1.0
decorator==4.4.0
defusedxml==0.6.0
distributed==2.1.0
docutils==0.14
entrypoints==0.3
et-xmlfile==1.0.1
fastcache==1.1.0
filelock==3.0.12
Flask==1.1.1
future==0.17.1
gevent==1.4.0
glob2==0.7
greenlet==0.4.15
h5py==2.9.0
heapdict==1.0.0
html5lib==1.0.1
idna==2.8
ijson==2.4
imageio==2.5.0
imagesize==1.1.0
importlib-metadata==0.17
ipykernel==5.1.1
ipython==7.6.1
ipython-genutils==0.2.0
ipywidgets==7.5.0
isort==4.3.21
itsdangerous==1.1.0
jdcal==1.4.1
jedi==0.13.3
Jinja2==2.10.1
jmespath==0.9.4
joblib==0.13.2
json5==0.8.4
jsonschema==3.0.1
jupyter==1.0.0
jupyter-client==5.3.1
jupyter-console==6.0.0
jupyter-core==4.5.0
jupyterlab==1.0.2
jupyterlab-server==1.0.0
Keras==2.2.4
Keras-Applications==1.0.8
Keras-Preprocessing==1.1.0
keyring==18.0.0
kiwisolver==1.1.0
lazy-object-proxy==1.4.1
libarchive-c==2.8
llvmlite==0.29.0
locket==0.2.0
lxml==4.3.4
MarkupSafe==1.1.1
matplotlib==3.1.0
mccabe==0.6.1
menuinst==1.4.16
mistune==0.8.4
mkl-fft==1.0.12
mkl-random==1.0.2
mkl-service==2.0.2
mock==3.0.5
more-itertools==7.0.0
mpmath==1.1.0
msgpack==0.6.1
multipledispatch==0.6.0
navigator-updater==0.2.1
nbconvert==5.5.0
nbformat==4.4.0
networkx==2.3
nltk==3.4.4
nose==1.3.7
notebook==6.0.0
numba==0.44.1
numexpr==2.6.9
numpy==1.16.4
numpydoc==0.9.1
olefile==0.46
openpyxl==2.6.2
packaging==19.0
pandas==0.24.2
pandocfilters==1.4.2
parso==0.5.0
partd==1.0.0
path.py==12.0.1
pathlib2==2.3.4
patsy==0.5.1
pep8==1.7.1
pickleshare==0.7.5
Pillow==6.1.0
pkginfo==1.5.0.1
pluggy==0.12.0
ply==3.11
prometheus-client==0.7.1
prompt-toolkit==2.0.9
psutil==5.6.3
py==1.8.0
pycodestyle==2.5.0
pycosat==0.6.3
pycparser==2.19
pycrypto==2.6.1
pycryptodomex==3.8.2
pycurl==7.43.0.3
pyflakes==2.1.1
Pygments==2.4.2
PyJWT==1.7.1
pylint==2.3.1
pyodbc==4.0.26
pyOpenSSL==19.0.0
pyparsing==2.4.0
pyreadline==2.1
pyrsistent==0.14.11
PySocks==1.7.0
pytest==5.0.1
pytest-arraydiff==0.3
pytest-astropy==0.5.0
pytest-doctestplus==0.3.0
pytest-openfiles==0.3.2
pytest-remotedata==0.3.1
python-dateutil==2.8.0
pytz==2019.1
PyWavelets==1.0.3
pywin32==223
pywinpty==0.5.5
PyYAML==5.1.1
pyzmq==18.0.0
QtAwesome==0.5.7
qtconsole==4.5.1
QtPy==1.8.0
requests==2.22.0
rope==0.14.0
ruamel-yaml==0.15.46
s3transfer==0.2.1
scikit-image==0.15.0
scikit-learn==0.21.2
scipy==1.2.1
seaborn==0.9.0
Send2Trash==1.5.0
simplegeneric==0.8.1
singledispatch==3.4.0.3
six==1.12.0
sklearn==0.0
snowballstemmer==1.9.0
snowflake==0.0.3
snowflake-connector-python==1.8.7
sortedcollections==1.1.2
sortedcontainers==2.1.0
soupsieve==1.8
Sphinx==2.1.2
sphinxcontrib-applehelp==1.0.1
sphinxcontrib-devhelp==1.0.1
sphinxcontrib-htmlhelp==1.0.2
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==1.0.2
sphinxcontrib-serializinghtml==1.1.3
sphinxcontrib-websupport==1.1.2
spyder==3.3.6
spyder-kernels==0.5.1
SQLAlchemy==1.3.5
statsmodels==0.10.0
sympy==1.4
tables==3.5.2
tblib==1.4.0
terminado==0.8.2
testpath==0.4.2
toolz==0.10.0
tornado==6.0.3
tqdm==4.32.1
traitlets==4.3.2
unicodecsv==0.14.1
urllib3==1.24.2
wcwidth==0.1.7
webencodings==0.5.1
Werkzeug==0.15.4
widgetsnbextension==3.5.0
win-inet-pton==1.1.0
win-unicode-console==0.5
wincertstore==0.2
wrapt==1.11.2
xlrd==1.2.0
XlsxWriter==1.1.8
xlwings==0.15.8
xlwt==1.3.0
zict==1.0.0
zipp==0.5.1**

4. What did you do?
If possible, provide a recipe for reproducing the error.
A complete runnable program is good.

There was no error I ran the program last time(last Friday, June 12) but ran the exact same program(no code change since then) a few days ago and got the error below(#6).

**Exception has occurred: OperationalError
250003: Failed to execute request: 'SSLSocket' object has no attribute 'connection'**

I can't provide the script for the whole program here but I am sure that it's due to Snowflake changes.

Since many libraries for this program are dependent on each other, I've been very careful to upgrade any library here. However, as a test, I upgraded urllib3(urllib3-1.25.9) and snowflake-connector-python(2.0.3) and got another type of error(below).

**Exception has occurred: TypeError
ssl_wrap_socket_with_ocsp() got an unexpected keyword argument 'key_password'**

5. What did you expect to see?

no error/ run the app locally

6. What did you see instead?

**Exception has occurred: OperationalError
250003: Failed to execute request: 'SSLSocket' object has no attribute 'connection'**

7. Can you set logging to DEBUG and collect the logs?

**[The log from the original version]**
The log thread never ended(infinite). I am copying and pasting a part of that log.

2020-06-18 09:54:45,844 - Thread-50 ocsp_asn1crypto.py:183 - is_valid_time() - DEBUG - Certificate is attached in Basic OCSP Response
2020-06-18 09:54:45,845 - Thread-51 ocsp_asn1crypto.py:189 - is_valid_time() - DEBUG - Valid Not After: 2021-04-02 17:13:33+00:00
2020-06-18 09:54:45,858 - Thread-48 ocsp_asn1crypto.py:183 - is_valid_time() - DEBUG - Certificate is attached in Basic OCSP Response
2020-06-18 09:54:45,858 - Thread-49 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +3456(s)
2020-06-18 09:54:45,859 - Thread-50 ocsp_asn1crypto.py:185 - is_valid_time() - DEBUG - Verifying the attached certificate is signed by the issuer
2020-06-18 09:54:45,860 - Thread-51 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +3456(s)
2020-06-18 09:54:45,860 - Thread-48 ocsp_asn1crypto.py:185 - is_valid_time() - DEBUG - Verifying the attached certificate is signed by the issuer
2020-06-18 09:54:45,864 - Thread-49 ocsp_asn1crypto.py:183 - is_valid_time() - DEBUG - Certificate is attached in Basic OCSP Response
2020-06-18 09:54:45,871 - Thread-50 ocsp_asn1crypto.py:189 - is_valid_time() - DEBUG - Valid Not After: 2021-04-02 16:58:09+00:00
2020-06-18 09:54:45,875 - Thread-51 ocsp_asn1crypto.py:183 - is_valid_time() - DEBUG - Certificate is attached in Basic OCSP Response
2020-06-18 09:54:45,876 - Thread-48 ocsp_asn1crypto.py:189 - is_valid_time() - DEBUG - Valid Not After: 2021-04-02 16:58:09+00:00
2020-06-18 09:54:45,877 - Thread-49 ocsp_asn1crypto.py:185 - is_valid_time() - DEBUG - Verifying the attached certificate is signed by the issuer
2020-06-18 09:54:45,879 - Thread-50 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +3456(s)
2020-06-18 09:54:45,886 - Thread-51 ocsp_asn1crypto.py:185 - is_valid_time() - DEBUG - Verifying the attached certificate is signed by the issuer
2020-06-18 09:54:45,888 - Thread-48 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +3456(s)
2020-06-18 09:54:45,888 - Thread-49 ocsp_asn1crypto.py:189 - is_valid_time() - DEBUG - Valid Not After: 2021-04-02 16:58:09+00:00
2020-06-18 09:54:45,892 - Thread-50 ocsp_asn1crypto.py:183 - is_valid_time() - DEBUG - Certificate is attached in Basic OCSP Response
2020-06-18 09:54:45,893 - Thread-51 ocsp_asn1crypto.py:189 - is_valid_time() - DEBUG - Valid Not After: 2021-04-02 16:58:09+00:00
2020-06-18 09:54:45,897 - Thread-48 ocsp_asn1crypto.py:183 - is_valid_time() - DEBUG - Certificate is attached in Basic OCSP Response
2020-06-18 09:54:45,905 - Thread-49 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +3456(s)
2020-06-18 09:54:45,906 - Thread-50 ocsp_asn1crypto.py:185 - is_valid_time() - DEBUG - Verifying the attached certificate is signed by the issuer
2020-06-18 09:54:45,907 - Thread-51 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +3456(s)
2020-06-18 09:54:45,907 - Thread-48 ocsp_asn1crypto.py:185 - is_valid_time() - DEBUG - Verifying the attached certificate is signed by the issuer
2020-06-18 09:54:45,911 - Thread-49 ocsp_asn1crypto.py:183 - is_valid_time() - DEBUG - Certificate is attached in Basic OCSP Response
2020-06-18 09:54:45,912 - Thread-50 ocsp_asn1crypto.py:189 - is_valid_time() - DEBUG - Valid Not After: 2021-04-02 16:58:09+00:00
2020-06-18 09:54:45,924 - Thread-51 ocsp_asn1crypto.py:183 - is_valid_time() - DEBUG - Certificate is attached in Basic OCSP Response
2020-06-18 09:54:45,925 - Thread-48 ocsp_asn1crypto.py:189 - is_valid_time() - DEBUG - Valid Not After: 2021-04-02 16:58:09+00:00
2020-06-18 09:54:45,925 - Thread-49 ocsp_asn1crypto.py:185 - is_valid_time() - DEBUG - Verifying the attached certificate is signed by the issuer
2020-06-18 09:54:45,926 - Thread-50 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +3456(s)
2020-06-18 09:54:45,927 - Thread-51 ocsp_asn1crypto.py:185 - is_valid_time() - DEBUG - Verifying the attached certificate is signed by the issuer
2020-06-18 09:54:45,928 - Thread-48 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +3456(s)
2020-06-18 09:54:45,937 - Thread-49 ocsp_asn1crypto.py:189 - is_valid_time() - DEBUG - Valid Not After: 2021-04-02 16:58:09+00:00
2020-06-18 09:54:45,939 - Thread-50 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:45,939 - Thread-51 ocsp_asn1crypto.py:189 - is_valid_time() - DEBUG - Valid Not After: 2021-04-02 16:58:09+00:00
2020-06-18 09:54:45,940 - Thread-48 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:45,941 - Thread-49 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +3456(s)
2020-06-18 09:54:45,942 - Thread-50 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:45,943 - Thread-51 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +3456(s)
2020-06-18 09:54:45,944 - Thread-48 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:45,945 - Thread-49 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:45,954 - Thread-50 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:45,957 - Thread-51 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:45,957 - Thread-48 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:45,958 - Thread-49 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:45,961 - Thread-50 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:45,961 - Thread-51 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:45,972 - Thread-48 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:45,973 - Thread-49 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:45,973 - Thread-50 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:45,975 - Thread-51 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:45,976 - Thread-48 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:45,977 - Thread-49 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:45,978 - Thread-50 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:45,979 - Thread-51 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:45,989 - Thread-48 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:45,990 - Thread-49 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:45,991 - Thread-50 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:45,994 - Thread-51 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:45,994 - Thread-48 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:45,995 - Thread-49 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:46,005 - Thread-50 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:46,006 - Thread-51 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:46,008 - Thread-48 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:46,009 - Thread-49 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:46,013 - Thread-50 ocsp_asn1crypto.py:183 - is_valid_time() - DEBUG - Certificate is attached in Basic OCSP Response
2020-06-18 09:54:46,022 - Thread-51 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:46,026 - Thread-48 ocsp_asn1crypto.py:183 - is_valid_time() - DEBUG - Certificate is attached in Basic OCSP Response
2020-06-18 09:54:46,027 - Thread-49 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:46,037 - Thread-50 ocsp_asn1crypto.py:185 - is_valid_time() - DEBUG - Verifying the attached certificate is signed by the issuer
2020-06-18 09:54:46,038 - Thread-51 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:46,040 - Thread-48 ocsp_asn1crypto.py:185 - is_valid_time() - DEBUG - Verifying the attached certificate is signed by the issuer
2020-06-18 09:54:46,043 - Thread-49 ocsp_asn1crypto.py:183 - is_valid_time() - DEBUG - Certificate is attached in Basic OCSP Response
2020-06-18 09:54:46,044 - Thread-50 ocsp_asn1crypto.py:189 - is_valid_time() - DEBUG - Valid Not After: 2021-04-02 17:08:26+00:00
2020-06-18 09:54:46,048 - Thread-51 ocsp_asn1crypto.py:183 - is_valid_time() - DEBUG - Certificate is attached in Basic OCSP Response
2020-06-18 09:54:46,054 - Thread-48 ocsp_asn1crypto.py:189 - is_valid_time() - DEBUG - Valid Not After: 2021-04-02 17:08:26+00:00
2020-06-18 09:54:46,055 - Thread-49 ocsp_asn1crypto.py:185 - is_valid_time() - DEBUG - Verifying the attached certificate is signed by the issuer
2020-06-18 09:54:46,056 - Thread-50 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +3456(s)
2020-06-18 09:54:46,056 - Thread-51 ocsp_asn1crypto.py:185 - is_valid_time() - DEBUG - Verifying the attached certificate is signed by the issuer
2020-06-18 09:54:46,057 - Thread-48 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +3456(s)
2020-06-18 09:54:46,058 - Thread-49 ocsp_asn1crypto.py:189 - is_valid_time() - DEBUG - Valid Not After: 2021-04-02 17:08:26+00:00
2020-06-18 09:54:46,063 - Thread-50 ocsp_asn1crypto.py:183 - is_valid_time() - DEBUG - Certificate is attached in Basic OCSP Response
2020-06-18 09:54:46,070 - Thread-51 ocsp_asn1crypto.py:189 - is_valid_time() - DEBUG - Valid Not After: 2021-04-02 17:08:26+00:00
2020-06-18 09:54:46,074 - Thread-48 ocsp_asn1crypto.py:183 - is_valid_time() - DEBUG - Certificate is attached in Basic OCSP Response
2020-06-18 09:54:46,075 - Thread-49 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +3456(s)
2020-06-18 09:54:46,075 - Thread-50 ocsp_asn1crypto.py:185 - is_valid_time() - DEBUG - Verifying the attached certificate is signed by the issuer
2020-06-18 09:54:46,076 - Thread-51 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +3456(s)
2020-06-18 09:54:46,077 - Thread-48 ocsp_asn1crypto.py:185 - is_valid_time() - DEBUG - Verifying the attached certificate is signed by the issuer
2020-06-18 09:54:46,081 - Thread-49 ocsp_asn1crypto.py:183 - is_valid_time() - DEBUG - Certificate is attached in Basic OCSP Response
2020-06-18 09:54:46,088 - Thread-50 ocsp_asn1crypto.py:189 - is_valid_time() - DEBUG - Valid Not After: 2021-04-02 17:08:26+00:00
2020-06-18 09:54:46,093 - Thread-51 ocsp_asn1crypto.py:183 - is_valid_time() - DEBUG - Certificate is attached in Basic OCSP Response
2020-06-18 09:54:46,093 - Thread-48 ocsp_asn1crypto.py:189 - is_valid_time() - DEBUG - Valid Not After: 2021-04-02 17:08:26+00:00
2020-06-18 09:54:46,093 - Thread-49 ocsp_asn1crypto.py:185 - is_valid_time() - DEBUG - Verifying the attached certificate is signed by the issuer
2020-06-18 09:54:46,094 - Thread-50 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +3456(s)
2020-06-18 09:54:46,094 - Thread-51 ocsp_asn1crypto.py:185 - is_valid_time() - DEBUG - Verifying the attached certificate is signed by the issuer
2020-06-18 09:54:46,105 - Thread-48 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +3456(s)
2020-06-18 09:54:46,105 - Thread-49 ocsp_asn1crypto.py:189 - is_valid_time() - DEBUG - Valid Not After: 2021-04-02 17:08:26+00:00
2020-06-18 09:54:46,107 - Thread-50 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:46,107 - Thread-51 ocsp_asn1crypto.py:189 - is_valid_time() - DEBUG - Valid Not After: 2021-04-02 17:08:26+00:00
2020-06-18 09:54:46,109 - Thread-48 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:46,109 - Thread-49 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +3456(s)
2020-06-18 09:54:46,111 - Thread-50 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:46,111 - Thread-51 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +3456(s)
2020-06-18 09:54:46,112 - Thread-48 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:46,121 - Thread-49 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:46,124 - Thread-50 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:46,125 - Thread-51 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:46,125 - Thread-48 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:46,126 - Thread-49 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:46,127 - Thread-50 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:46,138 - Thread-51 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)
2020-06-18 09:54:46,139 - Thread-48 ocsp_snowflake.py:1207 - _is_validaity_range() - DEBUG - Tolerable Validity range for OCSP response: +6021(s)

**[The log after I upgraded urllib3 and snowflake-connector-python]**

2020-06-18 09:47:35,938 - MainThread connection.py:178 - __init__() - INFO - Snowflake Connector for Python Version: 2.0.3, Python Version: 3.7.2, Platform: Windows-10-10.0.18362-SP0
2020-06-18 09:47:35,940 - MainThread connection.py:478 - connect() - DEBUG - connect
2020-06-18 09:47:35,942 - MainThread connection.py:717 - __config() - DEBUG - __config
2020-06-18 09:47:35,961 - MainThread connection.py:792 - __config() - INFO - This connection is in OCSP Fail Open Mode. TLS Certificates would be checked for validity and revocation status. Any other Certificate Revocation related exceptions or OCSP Responder failures would be disregarded in favor of connectivity.
2020-06-18 09:47:35,964 - MainThread converter.py:140 - __init__() - DEBUG - use_numpy: False
2020-06-18 09:47:35,977 - MainThread converter_issue23517.py:27 - __init__() - DEBUG - initialized
2020-06-18 09:47:35,979 - MainThread connection.py:643 - __open_connection() - DEBUG - REST API object was created: qv34892.east-us-2.azure.snowflakecomputing.com:443
2020-06-18 09:47:35,980 - MainThread auth.py:129 - authenticate() - DEBUG - authenticate
2020-06-18 09:47:35,980 - MainThread auth.py:153 - authenticate() - DEBUG - assertion content: *********
2020-06-18 09:47:35,990 - MainThread auth.py:165 - authenticate() - DEBUG - account=qv34892, user=ACCRUENTBIGDATA, database=HC_DEV, schema=PRESENTATION, warehouse=None, role=None, request_id=c2d288fb-4f34-4dfb-a0cc-7d51f19c3a87
2020-06-18 09:47:35,992 - MainThread auth.py:191 - authenticate() - DEBUG - body['data']: {'CLIENT_APP_ID': 'PythonConnector', 'CLIENT_APP_VERSION': '2.0.3', 'SVN_REVISION': None, 'ACCOUNT_NAME': 'qv34892', 'LOGIN_NAME': 'ACCRUENTBIGDATA', 'CLIENT_ENVIRONMENT': {'APPLICATION': 'PythonConnector', 'OS': 'Windows', 'OS_VERSION': 'Windows-10-10.0.18362-SP0', 'PYTHON_VERSION': '3.7.2', 'PYTHON_RUNTIME': 'CPython', 'PYTHON_COMPILER': 'MSC v.1916 64 bit (AMD64)', 'OCSP_MODE': 'FAIL_OPEN'}, 'SESSION_PARAMETERS': {'CLIENT_SESSION_KEEP_ALIVE_HEARTBEAT_FREQUENCY': 900, 'CLIENT_PREFETCH_THREADS': 4}}
2020-06-18 09:47:35,994 - MainThread network.py:940 - _use_requests_session() - DEBUG - Active requests sessions: 1, idle: 0
2020-06-18 09:47:35,995 - MainThread network.py:639 - _request_exec_wrapper() - DEBUG - remaining request timeout: 120, retry cnt: 1
2020-06-18 09:47:35,996 - MainThread network.py:778 - _request_exec() - DEBUG - socket timeout: 60
2020-06-18 09:47:36,114 - MainThread network.py:953 - _use_requests_session() - DEBUG - Active requests sessions: 0, idle: 1

**Comment by sfc-gh-mkeller** [comment]
Seems like a duplicate of #324

**Comment by sfc-gh-mkeller** [comment]
Your second issue:
```
Exception has occurred: TypeError
ssl_wrap_socket_with_ocsp() got an unexpected keyword argument 'key_password'
```
Was fixed in our 2.1.0 release, in this commit: https://github.com/snowflakedb/snowflake-connector-python/commit/6c84a997a5e9603542066678c6428d92cc7ccd88

**Comment by sfc-gh-mkeller** [comment]
Closing based on discussion in #324
